### PR TITLE
Fixed internal server error on importing zip

### LINKED ISSUE #3672: Exception object does not have code attribute in api/helpers/import_helpers.py
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

**Current behavior:**
`'InvalidRequestError' object has no attribute 'code'` error when trying to import event from zip file

image

**Steps to reproduce:**
Manage Events > Import > Upload a zip file 

**Related code:**
https://github.com/fossasia/open-event-orga-server/blob/development/app/api/helpers/import_helpers.py#L113

### PR DESCRIPTION
**Pull request by enigmaeth**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 

Fixed internal server error on importing zip files.

Fixes #3672 and #3585 .

### CODE REVIEW

File: app/api/helpers/import_helpers.py [hunk 1]
```diff
@@ -110,8 +110,9 @@ def make_error(uploaded_file, er=None, id_=None):
     if id_ is not None:
         istr = '%s, ID %s' % (istr, id_)
     er.message = '%s, %s' % (istr, er.message)
-    if not er.code:
-        er.code = 500
+    if hasattr(er, 'code'):
+        if not er.code:
```
Comment (poush) [inline app/api/helpers/import_helpers.py]: May be you can try it as 
``` if hasattr(er, 'code') and not er.code: ```

Comment (enigmaeth) [inline app/api/helpers/import_helpers.py]: Thanks. Changed.

File: app/api/helpers/import_helpers.py [hunk 2]
```diff
@@ -110,8 +110,8 @@ def make_error(uploaded_file, er=None, id_=None):
     if id_ is not None:
         istr = '%s, ID %s' % (istr, id_)
     er.message = '%s, %s' % (istr, er.message)
-    if not er.code:
-        er.code = 500
+    if hasattr(er, 'code') and not er.code:
+            er.code = 500
```
Comment (poush) [inline app/api/helpers/import_helpers.py]: I think indentation need to be reduced by one tab.

Comment (enigmaeth) [inline app/api/helpers/import_helpers.py]: Sorry, missed it!

File: app/api/events.py [hunk 3]
```diff
@@ -168,7 +168,7 @@ def create(self, data, url):
         save_to_db(new_event, "Event saved")
         # set organizer
         role = Role.query.filter_by(name=ORGANIZER).first()
-        uer = UsersEventsRoles(g.user, new_event, role)
+        uer = UsersEventsRoles(user_id=creator_id, event=new_event, role=role)
```
Comment (aviaryan) [inline app/api/events.py]: This breaks the Events API because at all other places where `EventDAO.create` is used, no creator_id is being passed. example - `EventList.post` method.

File: app/api/helpers/import_helpers.py [hunk 4]
```diff
@@ -294,7 +294,7 @@ def create_service_from_json(task_handle, data, srv, event_id, service_ids=None)
     return ids
 
 
-def import_event_json(task_handle, zip_path):
+def import_event_json(zip_path, task_handle, creator_id):
```
Comment (aviaryan) [inline app/api/helpers/import_helpers.py]: This is unnecessary diff. There was no need to change the order here. Just add `creator_id` as last param.

File: app/api/helpers/import_helpers.py [hunk 5]
```diff
@@ -317,7 +317,7 @@ def import_event_json(task_handle, zip_path):
         _, data = _trim_id(data)
         srv = ('event', EventDAO)
         data = _delete_fields(srv, data)
-        new_event = EventDAO.create(data, 'dont')[0]
+        new_event = EventDAO.create(data, 'dont', creator_id,)[0]
```
Comment (aviaryan) [inline app/api/helpers/import_helpers.py]: Trailing comma here.

File: app/api/helpers/import_helpers.py (hunk 4 above)
Comment (enigmaeth) [inline app/api/helpers/import_helpers.py]: This was done because in the other format,this the order of arguments.

File: app/api/events.py (hunk 3 above)
Comment (enigmaeth) [inline app/api/events.py]: The error which we were getting `User instance already exists` was because both were being passed: `g.user.id` [here](https://github.com/fossasia/open-event-orga-server/blob/development/app/api/imports.py#L33) and `g.user` [here](https://github.com/fossasia/open-event-orga-server/blob/development/app/api/events.py#L171). Passing the creator_id that was passed while getting the celery task solves the issue.

I am not much sure about how we can incorporate this without breaking the Event API. Suggestions please!

Comment (enigmaeth) [inline app/api/events.py]: Making `creator_id` an optional argument in ```EventDAO.create(self, data, url)``` would do?
[Code](https://github.com/fossasia/open-event-orga-server/blob/development/app/api/events.py#L158)

File: app/api/helpers/import_helpers.py (hunk 4 above)
Comment (enigmaeth) [inline app/api/helpers/import_helpers.py]:
```
elif source_type == 'pentabarf':
        new_event = PentabarfImporter.import_data(file_path=file_path, task_handle=task_handle, creator_id=creator_id)
elif source_type == 'ical':
        new_event = ICalImporter.import_data(file_path=file_path, task_handle=task_handle, creator_id=creator_id)
```

File: app/api/events.py (hunk 3 above)
Comment (aviaryan) [inline app/api/events.py]: What you can do is check if creator_id is set or not. If not (i.e. None), use `g.user.id` in that place. This won't break the EventList.post method.

Comment (enigmaeth) [inline app/api/events.py]: I did this :
In UsersEventsRoles [here](https://github.com/fossasia/open-event-orga-server/blob/development/app/models/users_events_roles.py#L17
) I am passing `g.user.id` to `user_id`, which is the same as passing `creator_id` as param.
So we don't have to add creator_id anywhere now.

Comment (aviaryan) [inline app/api/events.py]: @enigmaeth Where did you do that? Please reflect the changes in the code.

Comment (enigmaeth) [inline app/api/events.py]: I haven't pushed it yet. I am asking if this is a good way? We don't have to pass creator_id for zip imports.

Comment (enigmaeth) [inline app/api/events.py]: @aviaryan please review with the changes made.
